src/main/java/com/kenny/openimgur/widget/GalleryWidgetProvider.java

```diff
@@ -0,0 +1,134 @@
+package com.kenny.openimgur.widget;
+
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.text.TextUtils;
+import android.widget.RemoteViews;
+
+import com.kenny.openimgur.R;
+
+/**
+ * Widget provider which tells the launcher to update the widgets
+ * Created by John on 4/3/2015.
+ */
+public class GalleryWidgetProvider extends AppWidgetProvider {
+
+    public static final String ACTION_NEXT = "ACTION_NEXT";
+    public static final String ACTION_PREVIOUS = "ACTION_PREVIOUS";
+    public static final String ACTION_VIEW = "ACTION_VIEW";
+
+    public static final String EXTRA_ITEM = "EXTRA_ITEM";
+
+    @Override
+    public void onDeleted(Context context, int[] appWidgetIds) {
+        super.onDeleted(context, appWidgetIds);
+    }
+
+    @Override
+    public void onDisabled(Context context) {
+        super.onDisabled(context);
+    }
+
+    @Override
+    public void onEnabled(Context context) {
+        super.onEnabled(context);
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getAction().equals(ACTION_NEXT)) {
+            RemoteViews rv = new RemoteViews(context.getPackageName(),
+                    R.layout.widget_layout);
+
+            rv.showNext(R.id.widget_list);
+
+            AppWidgetManager.getInstance(context).partiallyUpdateAppWidget(
+                    intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,
+                            AppWidgetManager.INVALID_APPWIDGET_ID), rv);
+        } else if (intent.getAction().equals(ACTION_PREVIOUS)) {
+            RemoteViews rv = new RemoteViews(context.getPackageName(),
+                    R.layout.widget_layout);
+
+            rv.showPrevious(R.id.widget_list);
+
+            AppWidgetManager.getInstance(context).partiallyUpdateAppWidget(
+                    intent.getIntExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,
+                            AppWidgetManager.INVALID_APPWIDGET_ID), rv);
+        } else if (intent.getAction().equals(ACTION_VIEW)) {
+            Intent viewIntent = new Intent(Intent.ACTION_VIEW);
```

Comment:
Would like to open the gallery page when tapping on an image, not a fullscreen view of the image.